.github/workflows/container.yml

```diff
@@ -0,0 +1,79 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# (c) Copyright 2023 Advanced Micro Devices, Inc.
+name: "Container Image"
+
+on:
+  workflow_dispatch:
+    inputs:
+      registry-prefix:
+        description: Registry host and org
+        type: string
+      sfptpd-version:
+        description: sfptpd version
+        type: string
+      ubi-image:
+        description: UBI_IMAGE
+        type: string
+  push:
+    branches: [ master ]
+
+permissions:
+  contents: read
+  packages: write
+jobs:
+  container-build:
+    name: Container Build
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    env:
+      REGISTRY_PREFIX: ${{ inputs.registry-prefix || '' }}
+      SFPTPD_VERSION: ${{ inputs.sfptpd-version || '' }}
+      UBI_IMAGE: ${{ inputs.ubi-image || 'registry.access.redhat.com/ubi9-minimal:9.2' }}
+    steps:
+
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - name: Metadata - env
+      run: |
+        repo="${GITHUB_REPOSITORY@L}"
+        echo "REGISTRY_PREFIX=${REGISTRY_PREFIX:-ghcr.io/$repo}" >> "$GITHUB_ENV"
+
+        versioning_output=$(./scripts/sfptpd_versioning read)
```

Comment:
What about `sfptpd_versioning derive`? Is it because the output (which is Fedora/Debian package version friendly) is not docker-friendly? E.g. use of `~` to order pre-release versions before releases that don't have suffix?

Reply:
I definitely wouldn't have both - the main point about the derived suffix is to distinguish from an official git-tagged release. So I'm mostly interested that artifacts don't look like one unless they come from a commit which is git-tagged.

I don't quite follow everything that is happening here. I'm happy for the string not to match what we would get for, say, an RPM build - if docker doesn't use ~ then we might as well go with the docker flow and I'm happy for something appropriate to be constructed instead of using the versioning script.

I can see that this `-dev` suffix achieves what I'm mostly concerned about, so I'm happy about that and happy for whatever simplifications flow from that. What happens if there is a git-tag for this commit? In that case I would not expect there to be a `-dev` suffix.